src/main/java/com/mattworzala/canary/server/givemeahome/Structure.java

```diff
@@ -1,16 +1,112 @@
 package com.mattworzala.canary.server.givemeahome;
 
+import com.google.gson.annotations.Expose;
 import net.minestom.server.coordinate.Vec;
+import net.minestom.server.instance.block.Block;
+import net.minestom.server.instance.block.BlockSetter;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 /**
  * This will be replaced eventually
- *
+ * <p>
  * todo(alex)
  */
 public class Structure {
 
+    @Expose
+    private String id;
+    @Expose
+    private Vec size;
+
+    @Expose
+    Map<Integer, Block> blockmap;
+
+    @Expose
+    List<BlockDef> blocks;
+
+    public record BlockDef(int blockId, int blockCount) {
+    }
+
+
+    public Structure(String id, int sizeX, int sizeY, int sizeZ) {
+        this.id = id;
+        size = new Vec(sizeX, sizeY, sizeZ);
+
+        this.blockmap = new HashMap<>();
+        this.blocks = new ArrayList<>();
+    }
+
+    public void putInBlockMap(int index, Block block) {
+        blockmap.put(index, block);
+    }
+
+    public void addToBlockDefList(BlockDef blockDef) {
+        blocks.add(blockDef);
+    }
+
+    public void setBlocks(List<BlockDef> blockDefs) {
+        this.blocks = blockDefs;
+    }
+
+    public List<BlockDef> getBlocks() {
+        return this.blocks;
+    }
+
+    public Vec getSize() {
+        return this.size;
+    }
+
+    public int getSizeX() {
+        return (int) this.size.x();
+    }
+
+    public int getSizeY() {
+        return (int) this.size.y();
+    }
+
+    public int getSizeZ() {
+        return (int) this.size.z();
+    }
+
+    /**
+     * Index assumes an XZY packing order
+     *
+     * @param index
+     * @param block
+     * @param blockSetter
+     */
+    private void setBlockInBlockSetter(int index, @NotNull Block block, BlockSetter blockSetter) {
+        int x = index % this.getSizeX();
+        int z = index % (this.getSizeX() * this.getSizeZ()) / this.getSizeZ();
+        int y = index / (this.getSizeX() * this.getSizeZ());
+        assert x <= this.getSizeX();
```

Comment:
This is more a comment for me, but in java you need to provide a vm argument to enable assertions (`-ea`), im not sure if junit has them enabled. If not we should not use them since they will not be tested at runtime.